File: ProblemSet/44. 最小子数组/python/44. 最小子数组 贪心法.py
class Solution:
    """
    @param: nums: a list of integers
    @return: A integer indicate the sum of minimum subarray
    """
    def minSubArray(self, nums):
        minSum, cur = nums[0], nums[0]
        for i in range(1, len(nums)):
            cur = min(nums[i], nums[i] + cur)
            minSum = min(minSum, cur)
        return minSum

File: ProblemSet/44. 最小子数组/java/44. 最小子数组 贪心法.java
import java.util.List;

public class Solution {
    /*
     * @param nums: a list of integers
     * @return: A integer indicate the sum of minimum subarray
     */
    public int minSubArray(List<Integer> nums) {
        int minSum = nums.get(0), sum = nums.get(0);
        for (int i = 1; i < nums.size(); ++i) {
            sum = Math.min(nums.get(i), nums.get(i) + sum);
            minSum = Math.min(minSum, sum);
        }
        return minSum;
    }
}